src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -213,19 +213,20 @@ public static unsafe bool Ipv6StringToAddress(ReadOnlySpan<char> ipSpan, ushort*
                 long result = 0;
                 if (!string.IsNullOrEmpty(scopeId))
                 {
-                    if (scopeId.Length < 2)
+                    uint interfaceIndex = Interop.Sys.GetNativeIPInterfaceIndex(scopeId);
```

Comment:
Rather than going right to the DllImports, there should be a Pal class that exposes the method to be called here, and then that method would delegate to the appropriate DllImport for the appropriate type.  I don't think there's one yet for IPAddress, but you can see other examples in the project for, for example, SocketAddressPal.  If you think this makes sense there, you could just add it to that one; otherwise, define a new one.

Reply:
SocketAddressPal is used in 7 csprojs so I decided to put it to HostInformationPal, let me know if a separate Pal type is better.

Reply:
SocketAddressPal is used in 7 projects so I decided to create a new one